src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -245,6 +247,20 @@ private UnboundLambda BindAnonymousFunction(CSharpSyntaxNode syntax, DiagnosticB
                         continue;
                     }
 
+                    if (name == "_" && underscoreMeansDiscard)
```

Comment:
dumb question about underscore.  does it have to be literally written as `_`?  or can it be written as `\u005f`.  I ask because it's unclear to me if features should be checking hte token's ValueText or raw text for these sorts of checks.

Reply:
Good catch. Added some tests around that and aligned the behavior with other discards (ie. only the `_` counts as underscore).